SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -10,50 +14,70 @@ namespace SkietbaanBE.Controllers
     public class UserController : Controller
     {
         private ModelsContext _context;
-        public UserController(ModelsContext db) {
+        public UserController(ModelsContext db)
+        {
             _context = db;
         }
 
         //public UserController() { }
 
         // GET: api/User
         [HttpGet]
-        public IEnumerable<User> Get() {
+        public IEnumerable<User> Get()
+        {
             var users = _context.Users.ToArray<User>();
             return users;
         }
 
         // GET: api/User/5
         [HttpGet("{id}", Name = "Get")]
-        public User Get(int id) {
-            User user = _context.Users.Find(id);
+        public async Task<User> Get(int id)
+        {
+            User user = await _context.Users.FindAsync(id);
             return user;
         }
 
-        [HttpGet("test/{ret}", Name = "Test")]
-        public string Test(string ret) {
-            return $"Returned : {ret}";
-        }
-
         // POST: api/User
         [HttpPost]
-        public virtual void Post([FromBody] User user) {
-            _context.Add(user);
-            _context.SaveChanges();
+        public async Task<HttpResponseMessage> Post([FromBody] User user)
+        {
+            
+            if (ModelState.IsValid)
+            {
+                //get today's date and save it under user entry date
+                DateTime currentDate = new DateTime();
+                user.EntryDate = currentDate;
+
+                //Save User
+                await _context.AddAsync(user);
+                await _context.SaveChangesAsync();
+
+
+                return new HttpResponseMessage(HttpStatusCode.OK);
+            }
+            else
+            {
+                return new HttpResponseMessage(HttpStatusCode.BadRequest);
+            }
         }
 
         // PUT: api/User/5
         [HttpPut]
-        public void Put([FromBody] User user) {
-            _context.Users.Update(user);
-            _context.SaveChanges();
-        }
+        public async Task<ActionResult> Put([FromBody] User user)
+        {
+            //error handling, check if client provided valid data
+            if (user == null)
+            {
+                return new BadRequestObjectResult("user cannot be null");
+            }
+            else if (Get(user.Id) == null)
+            {
+                return new OkObjectResult("user does not exist");
+            }
 
-        // DELETE: api/ApiWithActions/5
-        [HttpDelete]
-        public void Delete(User user) {
-            _context.Users.Remove(user);
-            _context.SaveChanges();
+            _context.Users.Update(user);
```

Comment:
Where is your check if the user actually exists in the db? You have a check to ensure the user has an id but not if he actually exists

Reply:
missed it, resolving now